// modern-treasury/resources/payment-orders/payment-orders.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
var __createBinding =
  (this && this.__createBinding) ||
  (Object.create ?
    function (o, m, k, k2) {
      if (k2 === undefined) k2 = k;
      var desc = Object.getOwnPropertyDescriptor(m, k);
      if (!desc || ('get' in desc ? !m.__esModule : desc.writable || desc.configurable)) {
        desc = {
          enumerable: true,
          get: function () {
            return m[k];
          },
        };
      }
      Object.defineProperty(o, k2, desc);
    }
  : function (o, m, k, k2) {
      if (k2 === undefined) k2 = k;
      o[k2] = m[k];
    });
var __setModuleDefault =
  (this && this.__setModuleDefault) ||
  (Object.create ?
    function (o, v) {
      Object.defineProperty(o, 'default', { enumerable: true, value: v });
    }
  : function (o, v) {
      o['default'] = v;
    });
var __importStar =
  (this && this.__importStar) ||
  function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod)
        if (k !== 'default' && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
  };
Object.defineProperty(exports, '__esModule', { value: true });
exports.PaymentOrdersPage = exports.PaymentOrders = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const reversals_1 = require('./reversals.js');
const API = __importStar(require('./index.js'));
const core_2 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class PaymentOrders extends resource_1.APIResource {
  constructor() {
    super(...arguments);
    this.reversals = new reversals_1.Reversals(this.client);
  }
  /**
   * Create a new Payment Order
   */
  async create(params, options) {
    const { 'Content-Type': contentType, 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post(
      '/api/payment_orders',
      await (0, core_2.maybeMultipartFormRequestOptions)({
        body,
        ...options,
        headers: {
          'Content-Type':
            (contentType === null || contentType === void 0 ? void 0 : contentType.toString()) || '',
          'Idempotency-Key': idempotencyKey || '',
          ...(options === null || options === void 0 ? void 0 : options.headers),
        },
      }),
    );
  }
  /**
   * Get details on a single payment order
   */
  retrieve(id, options) {
    return this.get(`/api/payment_orders/${id}`, options);
  }
  update(id, body = {}, options) {
    if ((0, core_1.isRequestOptions)(body)) {
      return this.update(id, {}, body);
    }
    return this.patch(`/api/payment_orders/${id}`, { body, ...options });
  }
  list(query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/payment_orders', PaymentOrdersPage, { query, ...options });
  }
  /**
   * Create a new payment order asynchronously
   */
  createAsync(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/payment_orders/create_async', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
}
exports.PaymentOrders = PaymentOrders;
class PaymentOrdersPage extends pagination_1.Page {}
exports.PaymentOrdersPage = PaymentOrdersPage;
(function (PaymentOrders) {
  PaymentOrders.Reversals = API.Reversals;
  PaymentOrders.ReversalsPage = API.ReversalsPage;
})((PaymentOrders = exports.PaymentOrders || (exports.PaymentOrders = {})));
