// modern-treasury/resources/payment-orders/reversals.mjs
// File generated from our OpenAPI spec by Stainless.
import { APIResource } from 'modern-treasury/resource';
import { isRequestOptions } from 'modern-treasury/core';
import { Page } from 'modern-treasury/pagination';
export class Reversals extends APIResource {
  /**
   * Create a reversal for a payment order.
   */
  create(paymentOrderId, params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post(`/api/payment_orders/${paymentOrderId}/reversals`, {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * Get details on a single reversal of a payment order.
   */
  retrieve(paymentOrderId, reversalId, options) {
    return this.get(`/api/payment_orders/${paymentOrderId}/reversals/${reversalId}`, options);
  }
  list(paymentOrderId, query = {}, options) {
    if (isRequestOptions(query)) {
      return this.list(paymentOrderId, {}, query);
    }
    return this.getAPIList(`/api/payment_orders/${paymentOrderId}/reversals`, ReversalsPage, {
      query,
      ...options,
    });
  }
}
export class ReversalsPage extends Page {}
(function (Reversals) {})(Reversals || (Reversals = {}));
