// modern-treasury/resources/payment-orders/reversals.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.ReversalsPage = exports.Reversals = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class Reversals extends resource_1.APIResource {
  /**
   * Create a reversal for a payment order.
   */
  create(paymentOrderId, params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post(`/api/payment_orders/${paymentOrderId}/reversals`, {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * Get details on a single reversal of a payment order.
   */
  retrieve(paymentOrderId, reversalId, options) {
    return this.get(`/api/payment_orders/${paymentOrderId}/reversals/${reversalId}`, options);
  }
  list(paymentOrderId, query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list(paymentOrderId, {}, query);
    }
    return this.getAPIList(`/api/payment_orders/${paymentOrderId}/reversals`, ReversalsPage, {
      query,
      ...options,
    });
  }
}
exports.Reversals = Reversals;
class ReversalsPage extends pagination_1.Page {}
exports.ReversalsPage = ReversalsPage;
(function (Reversals) {})((Reversals = exports.Reversals || (exports.Reversals = {})));
