// modern-treasury/resources/internal-accounts/internal-accounts.mjs
// File generated from our OpenAPI spec by Stainless.
import { APIResource } from 'modern-treasury/resource';
import { isRequestOptions } from 'modern-treasury/core';
import { BalanceReports } from './balance-reports.mjs';
import * as API from './index.mjs';
import { Page } from 'modern-treasury/pagination';
export class InternalAccounts extends APIResource {
  constructor() {
    super(...arguments);
    this.balanceReports = new BalanceReports(this.client);
  }
  /**
   * create internal account
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/internal_accounts', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * get internal account
   */
  retrieve(id, options) {
    return this.get(`/api/internal_accounts/${id}`, options);
  }
  update(id, body = {}, options) {
    if (isRequestOptions(body)) {
      return this.update(id, {}, body);
    }
    return this.patch(`/api/internal_accounts/${id}`, { body, ...options });
  }
  list(query = {}, options) {
    if (isRequestOptions(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/internal_accounts', InternalAccountsPage, { query, ...options });
  }
}
export class InternalAccountsPage extends Page {}
(function (InternalAccounts) {
  InternalAccounts.BalanceReports = API.BalanceReports;
  InternalAccounts.BalanceReportsPage = API.BalanceReportsPage;
})(InternalAccounts || (InternalAccounts = {}));
