// modern-treasury/resources/account-details.mjs
// File generated from our OpenAPI spec by Stainless.
import { APIResource } from 'modern-treasury/resource';
import { isRequestOptions } from 'modern-treasury/core';
import { Page } from 'modern-treasury/pagination';
export class AccountDetails extends APIResource {
  /**
   * Create an account detail for an external account.
   */
  create(accountsType, accountId, params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post(`/api/${accountsType}/${accountId}/account_details`, {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * Get a single account detail for a single internal or external account.
   */
  retrieve(accountsType, accountId, id, options) {
    return this.get(`/api/${accountsType}/${accountId}/account_details/${id}`, options);
  }
  list(accountsType, accountId, query = {}, options) {
    if (isRequestOptions(query)) {
      return this.list(accountsType, accountId, {}, query);
    }
    return this.getAPIList(`/api/${accountsType}/${accountId}/account_details`, AccountDetailsPage, {
      query,
      ...options,
    });
  }
  /**
   * Delete a single account detail for an external account.
   */
  del(accountsType, accountId, id, options) {
    return this.delete(`/api/${accountsType}/${accountId}/account_details/${id}`, {
      ...options,
      headers: { Accept: '', ...(options === null || options === void 0 ? void 0 : options.headers) },
    });
  }
}
export class AccountDetailsPage extends Page {}
(function (AccountDetails) {})(AccountDetails || (AccountDetails = {}));
