// modern-treasury/resources/account-collection-flows.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.AccountCollectionFlowsPage = exports.AccountCollectionFlows = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class AccountCollectionFlows extends resource_1.APIResource {
  /**
   * create account_collection_flow
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/account_collection_flows', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  retrieve(id, query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.retrieve(id, {}, query);
    }
    const { 'Idempotency-Key': idempotencyKey } = query;
    return this.get(`/api/account_collection_flows/${id}`, {
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * update account_collection_flow
   */
  update(id, params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.patch(`/api/account_collection_flows/${id}`, {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  list(query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/account_collection_flows', AccountCollectionFlowsPage, {
      query,
      ...options,
    });
  }
}
exports.AccountCollectionFlows = AccountCollectionFlows;
class AccountCollectionFlowsPage extends pagination_1.Page {}
exports.AccountCollectionFlowsPage = AccountCollectionFlowsPage;
(function (AccountCollectionFlows) {})(
  (AccountCollectionFlows = exports.AccountCollectionFlows || (exports.AccountCollectionFlows = {})),
);
