// modern-treasury/resources/documents.mjs
// File generated from our OpenAPI spec by Stainless.
import { APIResource } from 'modern-treasury/resource';
import { isRequestOptions } from 'modern-treasury/core';
import { multipartFormRequestOptions } from 'modern-treasury/core';
import { Page } from 'modern-treasury/pagination';
export class Documents extends APIResource {
  /**
   * Create a document.
   */
  async create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post(
      '/api/documents',
      await multipartFormRequestOptions({
        body,
        ...options,
        headers: {
          'Idempotency-Key': idempotencyKey || '',
          ...(options === null || options === void 0 ? void 0 : options.headers),
        },
      }),
    );
  }
  /**
   * Get an existing document.
   */
  retrieve(id, options) {
    return this.get(`/api/documents/${id}`, options);
  }
  list(query = {}, options) {
    if (isRequestOptions(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/documents', DocumentsPage, { query, ...options });
  }
}
export class DocumentsPage extends Page {}
(function (Documents) {})(Documents || (Documents = {}));
