// modern-treasury/resources/documents.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.DocumentsPage = exports.Documents = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const core_2 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class Documents extends resource_1.APIResource {
  /**
   * Create a document.
   */
  async create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post(
      '/api/documents',
      await (0, core_2.multipartFormRequestOptions)({
        body,
        ...options,
        headers: {
          'Idempotency-Key': idempotencyKey || '',
          ...(options === null || options === void 0 ? void 0 : options.headers),
        },
      }),
    );
  }
  /**
   * Get an existing document.
   */
  retrieve(id, options) {
    return this.get(`/api/documents/${id}`, options);
  }
  list(query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/documents', DocumentsPage, { query, ...options });
  }
}
exports.Documents = Documents;
class DocumentsPage extends pagination_1.Page {}
exports.DocumentsPage = DocumentsPage;
(function (Documents) {})((Documents = exports.Documents || (exports.Documents = {})));
