// modern-treasury/resources/expected-payments.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.ExpectedPaymentsPage = exports.ExpectedPayments = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class ExpectedPayments extends resource_1.APIResource {
  /**
   * create expected payment
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/expected_payments', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * get expected payment
   */
  retrieve(id, options) {
    return this.get(`/api/expected_payments/${id}`, options);
  }
  update(id, body = {}, options) {
    if ((0, core_1.isRequestOptions)(body)) {
      return this.update(id, {}, body);
    }
    return this.patch(`/api/expected_payments/${id}`, { body, ...options });
  }
  list(query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/expected_payments', ExpectedPaymentsPage, { query, ...options });
  }
  /**
   * delete expected payment
   */
  del(id, options) {
    return this.delete(`/api/expected_payments/${id}`, options);
  }
}
exports.ExpectedPayments = ExpectedPayments;
class ExpectedPaymentsPage extends pagination_1.Page {}
exports.ExpectedPaymentsPage = ExpectedPaymentsPage;
(function (ExpectedPayments) {})(
  (ExpectedPayments = exports.ExpectedPayments || (exports.ExpectedPayments = {})),
);
