// modern-treasury/resources/external-accounts.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.ExternalAccountsPage = exports.ExternalAccounts = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class ExternalAccounts extends resource_1.APIResource {
  /**
   * create external account
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/external_accounts', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * show external account
   */
  retrieve(id, options) {
    return this.get(`/api/external_accounts/${id}`, options);
  }
  update(id, body = {}, options) {
    if ((0, core_1.isRequestOptions)(body)) {
      return this.update(id, {}, body);
    }
    return this.patch(`/api/external_accounts/${id}`, { body, ...options });
  }
  list(query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/external_accounts', ExternalAccountsPage, { query, ...options });
  }
  /**
   * delete external account
   */
  del(id, options) {
    return this.delete(`/api/external_accounts/${id}`, {
      ...options,
      headers: { Accept: '', ...(options === null || options === void 0 ? void 0 : options.headers) },
    });
  }
  completeVerification(id, params = {}, options) {
    if ((0, core_1.isRequestOptions)(params)) {
      return this.completeVerification(id, {}, params);
    }
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post(`/api/external_accounts/${id}/complete_verification`, {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * verify external account
   */
  verify(id, params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post(`/api/external_accounts/${id}/verify`, {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
}
exports.ExternalAccounts = ExternalAccounts;
class ExternalAccountsPage extends pagination_1.Page {}
exports.ExternalAccountsPage = ExternalAccountsPage;
(function (ExternalAccounts) {})(
  (ExternalAccounts = exports.ExternalAccounts || (exports.ExternalAccounts = {})),
);
