// modern-treasury/resources/incoming-payment-details.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.IncomingPaymentDetailsPage = exports.IncomingPaymentDetails = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class IncomingPaymentDetails extends resource_1.APIResource {
  /**
   * Get an existing Incoming Payment Detail.
   */
  retrieve(id, options) {
    return this.get(`/api/incoming_payment_details/${id}`, options);
  }
  update(id, body = {}, options) {
    if ((0, core_1.isRequestOptions)(body)) {
      return this.update(id, {}, body);
    }
    return this.patch(`/api/incoming_payment_details/${id}`, { body, ...options });
  }
  list(query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/incoming_payment_details', IncomingPaymentDetailsPage, {
      query,
      ...options,
    });
  }
  createAsync(params = {}, options) {
    if ((0, core_1.isRequestOptions)(params)) {
      return this.createAsync({}, params);
    }
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/simulations/incoming_payment_details/create_async', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
}
exports.IncomingPaymentDetails = IncomingPaymentDetails;
class IncomingPaymentDetailsPage extends pagination_1.Page {}
exports.IncomingPaymentDetailsPage = IncomingPaymentDetailsPage;
(function (IncomingPaymentDetails) {})(
  (IncomingPaymentDetails = exports.IncomingPaymentDetails || (exports.IncomingPaymentDetails = {})),
);
