// modern-treasury/resources/incoming-payment-details.mjs
// File generated from our OpenAPI spec by Stainless.
import { APIResource } from 'modern-treasury/resource';
import { isRequestOptions } from 'modern-treasury/core';
import { Page } from 'modern-treasury/pagination';
export class IncomingPaymentDetails extends APIResource {
  /**
   * Get an existing Incoming Payment Detail.
   */
  retrieve(id, options) {
    return this.get(`/api/incoming_payment_details/${id}`, options);
  }
  update(id, body = {}, options) {
    if (isRequestOptions(body)) {
      return this.update(id, {}, body);
    }
    return this.patch(`/api/incoming_payment_details/${id}`, { body, ...options });
  }
  list(query = {}, options) {
    if (isRequestOptions(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/incoming_payment_details', IncomingPaymentDetailsPage, {
      query,
      ...options,
    });
  }
  createAsync(params = {}, options) {
    if (isRequestOptions(params)) {
      return this.createAsync({}, params);
    }
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/simulations/incoming_payment_details/create_async', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
}
export class IncomingPaymentDetailsPage extends Page {}
(function (IncomingPaymentDetails) {})(IncomingPaymentDetails || (IncomingPaymentDetails = {}));
