// modern-treasury/resources/index.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
var __createBinding =
  (this && this.__createBinding) ||
  (Object.create ?
    function (o, m, k, k2) {
      if (k2 === undefined) k2 = k;
      var desc = Object.getOwnPropertyDescriptor(m, k);
      if (!desc || ('get' in desc ? !m.__esModule : desc.writable || desc.configurable)) {
        desc = {
          enumerable: true,
          get: function () {
            return m[k];
          },
        };
      }
      Object.defineProperty(o, k2, desc);
    }
  : function (o, m, k, k2) {
      if (k2 === undefined) k2 = k;
      o[k2] = m[k];
    });
var __exportStar =
  (this && this.__exportStar) ||
  function (m, exports) {
    for (var p in m)
      if (p !== 'default' && !Object.prototype.hasOwnProperty.call(exports, p))
        __createBinding(exports, m, p);
  };
Object.defineProperty(exports, '__esModule', { value: true });
exports.Returns =
  exports.ReturnObjectsPage =
  exports.PaymentReferences =
  exports.PaymentReferencesPage =
  exports.PaymentOrders =
  exports.PaymentOrdersPage =
  exports.PaymentFlows =
  exports.PaymentFlowsPage =
  exports.PaperItems =
  exports.PaperItemsPage =
  exports.LineItems =
  exports.LineItemsPage =
  exports.LedgerableEvents =
  exports.LedgerTransactions =
  exports.LedgerTransactionsPage =
  exports.LedgerEventHandlers =
  exports.LedgerEventHandlerListResponsesPage =
  exports.LedgerEntries =
  exports.LedgerEntriesPage =
  exports.LedgerAccountStatements =
  exports.LedgerAccountPayouts =
  exports.LedgerAccountPayoutsPage =
  exports.LedgerAccountCategories =
  exports.LedgerAccountCategoriesPage =
  exports.LedgerAccounts =
  exports.LedgerAccountsPage =
  exports.Ledgers =
  exports.LedgersPage =
  exports.Invoices =
  exports.InvoicesPage =
  exports.InternalAccounts =
  exports.InternalAccountsPage =
  exports.IncomingPaymentDetails =
  exports.IncomingPaymentDetailsPage =
  exports.ExternalAccounts =
  exports.ExternalAccountsPage =
  exports.ExpectedPayments =
  exports.ExpectedPaymentsPage =
  exports.Events =
  exports.EventsPage =
  exports.Documents =
  exports.DocumentsPage =
  exports.Counterparties =
  exports.CounterpartiesPage =
  exports.Connections =
  exports.ConnectionsPage =
  exports.AccountDetails =
  exports.AccountDetailsPage =
  exports.AccountCollectionFlows =
  exports.AccountCollectionFlowsPage =
    void 0;
exports.Webhooks =
  exports.VirtualAccounts =
  exports.VirtualAccountsPage =
  exports.Transactions =
  exports.TransactionsPage =
  exports.Validations =
  exports.RoutingDetails =
  exports.RoutingDetailsPage =
    void 0;
__exportStar(require('./shared.js'), exports);
var account_collection_flows_1 = require('./account-collection-flows.js');
Object.defineProperty(exports, 'AccountCollectionFlowsPage', {
  enumerable: true,
  get: function () {
    return account_collection_flows_1.AccountCollectionFlowsPage;
  },
});
Object.defineProperty(exports, 'AccountCollectionFlows', {
  enumerable: true,
  get: function () {
    return account_collection_flows_1.AccountCollectionFlows;
  },
});
var account_details_1 = require('./account-details.js');
Object.defineProperty(exports, 'AccountDetailsPage', {
  enumerable: true,
  get: function () {
    return account_details_1.AccountDetailsPage;
  },
});
Object.defineProperty(exports, 'AccountDetails', {
  enumerable: true,
  get: function () {
    return account_details_1.AccountDetails;
  },
});
var connections_1 = require('./connections.js');
Object.defineProperty(exports, 'ConnectionsPage', {
  enumerable: true,
  get: function () {
    return connections_1.ConnectionsPage;
  },
});
Object.defineProperty(exports, 'Connections', {
  enumerable: true,
  get: function () {
    return connections_1.Connections;
  },
});
var counterparties_1 = require('./counterparties.js');
Object.defineProperty(exports, 'CounterpartiesPage', {
  enumerable: true,
  get: function () {
    return counterparties_1.CounterpartiesPage;
  },
});
Object.defineProperty(exports, 'Counterparties', {
  enumerable: true,
  get: function () {
    return counterparties_1.Counterparties;
  },
});
var documents_1 = require('./documents.js');
Object.defineProperty(exports, 'DocumentsPage', {
  enumerable: true,
  get: function () {
    return documents_1.DocumentsPage;
  },
});
Object.defineProperty(exports, 'Documents', {
  enumerable: true,
  get: function () {
    return documents_1.Documents;
  },
});
var events_1 = require('./events.js');
Object.defineProperty(exports, 'EventsPage', {
  enumerable: true,
  get: function () {
    return events_1.EventsPage;
  },
});
Object.defineProperty(exports, 'Events', {
  enumerable: true,
  get: function () {
    return events_1.Events;
  },
});
var expected_payments_1 = require('./expected-payments.js');
Object.defineProperty(exports, 'ExpectedPaymentsPage', {
  enumerable: true,
  get: function () {
    return expected_payments_1.ExpectedPaymentsPage;
  },
});
Object.defineProperty(exports, 'ExpectedPayments', {
  enumerable: true,
  get: function () {
    return expected_payments_1.ExpectedPayments;
  },
});
var external_accounts_1 = require('./external-accounts.js');
Object.defineProperty(exports, 'ExternalAccountsPage', {
  enumerable: true,
  get: function () {
    return external_accounts_1.ExternalAccountsPage;
  },
});
Object.defineProperty(exports, 'ExternalAccounts', {
  enumerable: true,
  get: function () {
    return external_accounts_1.ExternalAccounts;
  },
});
var incoming_payment_details_1 = require('./incoming-payment-details.js');
Object.defineProperty(exports, 'IncomingPaymentDetailsPage', {
  enumerable: true,
  get: function () {
    return incoming_payment_details_1.IncomingPaymentDetailsPage;
  },
});
Object.defineProperty(exports, 'IncomingPaymentDetails', {
  enumerable: true,
  get: function () {
    return incoming_payment_details_1.IncomingPaymentDetails;
  },
});
var internal_accounts_1 = require('./internal-accounts/internal-accounts.js');
Object.defineProperty(exports, 'InternalAccountsPage', {
  enumerable: true,
  get: function () {
    return internal_accounts_1.InternalAccountsPage;
  },
});
Object.defineProperty(exports, 'InternalAccounts', {
  enumerable: true,
  get: function () {
    return internal_accounts_1.InternalAccounts;
  },
});
var invoices_1 = require('./invoices/invoices.js');
Object.defineProperty(exports, 'InvoicesPage', {
  enumerable: true,
  get: function () {
    return invoices_1.InvoicesPage;
  },
});
Object.defineProperty(exports, 'Invoices', {
  enumerable: true,
  get: function () {
    return invoices_1.Invoices;
  },
});
var ledgers_1 = require('./ledgers.js');
Object.defineProperty(exports, 'LedgersPage', {
  enumerable: true,
  get: function () {
    return ledgers_1.LedgersPage;
  },
});
Object.defineProperty(exports, 'Ledgers', {
  enumerable: true,
  get: function () {
    return ledgers_1.Ledgers;
  },
});
var ledger_accounts_1 = require('./ledger-accounts.js');
Object.defineProperty(exports, 'LedgerAccountsPage', {
  enumerable: true,
  get: function () {
    return ledger_accounts_1.LedgerAccountsPage;
  },
});
Object.defineProperty(exports, 'LedgerAccounts', {
  enumerable: true,
  get: function () {
    return ledger_accounts_1.LedgerAccounts;
  },
});
var ledger_account_categories_1 = require('./ledger-account-categories.js');
Object.defineProperty(exports, 'LedgerAccountCategoriesPage', {
  enumerable: true,
  get: function () {
    return ledger_account_categories_1.LedgerAccountCategoriesPage;
  },
});
Object.defineProperty(exports, 'LedgerAccountCategories', {
  enumerable: true,
  get: function () {
    return ledger_account_categories_1.LedgerAccountCategories;
  },
});
var ledger_account_payouts_1 = require('./ledger-account-payouts.js');
Object.defineProperty(exports, 'LedgerAccountPayoutsPage', {
  enumerable: true,
  get: function () {
    return ledger_account_payouts_1.LedgerAccountPayoutsPage;
  },
});
Object.defineProperty(exports, 'LedgerAccountPayouts', {
  enumerable: true,
  get: function () {
    return ledger_account_payouts_1.LedgerAccountPayouts;
  },
});
var ledger_account_statements_1 = require('./ledger-account-statements.js');
Object.defineProperty(exports, 'LedgerAccountStatements', {
  enumerable: true,
  get: function () {
    return ledger_account_statements_1.LedgerAccountStatements;
  },
});
var ledger_entries_1 = require('./ledger-entries.js');
Object.defineProperty(exports, 'LedgerEntriesPage', {
  enumerable: true,
  get: function () {
    return ledger_entries_1.LedgerEntriesPage;
  },
});
Object.defineProperty(exports, 'LedgerEntries', {
  enumerable: true,
  get: function () {
    return ledger_entries_1.LedgerEntries;
  },
});
var ledger_event_handlers_1 = require('./ledger-event-handlers.js');
Object.defineProperty(exports, 'LedgerEventHandlerListResponsesPage', {
  enumerable: true,
  get: function () {
    return ledger_event_handlers_1.LedgerEventHandlerListResponsesPage;
  },
});
Object.defineProperty(exports, 'LedgerEventHandlers', {
  enumerable: true,
  get: function () {
    return ledger_event_handlers_1.LedgerEventHandlers;
  },
});
var ledger_transactions_1 = require('./ledger-transactions/ledger-transactions.js');
Object.defineProperty(exports, 'LedgerTransactionsPage', {
  enumerable: true,
  get: function () {
    return ledger_transactions_1.LedgerTransactionsPage;
  },
});
Object.defineProperty(exports, 'LedgerTransactions', {
  enumerable: true,
  get: function () {
    return ledger_transactions_1.LedgerTransactions;
  },
});
var ledgerable_events_1 = require('./ledgerable-events.js');
Object.defineProperty(exports, 'LedgerableEvents', {
  enumerable: true,
  get: function () {
    return ledgerable_events_1.LedgerableEvents;
  },
});
var line_items_1 = require('./line-items.js');
Object.defineProperty(exports, 'LineItemsPage', {
  enumerable: true,
  get: function () {
    return line_items_1.LineItemsPage;
  },
});
Object.defineProperty(exports, 'LineItems', {
  enumerable: true,
  get: function () {
    return line_items_1.LineItems;
  },
});
var paper_items_1 = require('./paper-items.js');
Object.defineProperty(exports, 'PaperItemsPage', {
  enumerable: true,
  get: function () {
    return paper_items_1.PaperItemsPage;
  },
});
Object.defineProperty(exports, 'PaperItems', {
  enumerable: true,
  get: function () {
    return paper_items_1.PaperItems;
  },
});
var payment_flows_1 = require('./payment-flows.js');
Object.defineProperty(exports, 'PaymentFlowsPage', {
  enumerable: true,
  get: function () {
    return payment_flows_1.PaymentFlowsPage;
  },
});
Object.defineProperty(exports, 'PaymentFlows', {
  enumerable: true,
  get: function () {
    return payment_flows_1.PaymentFlows;
  },
});
var payment_orders_1 = require('./payment-orders/payment-orders.js');
Object.defineProperty(exports, 'PaymentOrdersPage', {
  enumerable: true,
  get: function () {
    return payment_orders_1.PaymentOrdersPage;
  },
});
Object.defineProperty(exports, 'PaymentOrders', {
  enumerable: true,
  get: function () {
    return payment_orders_1.PaymentOrders;
  },
});
var payment_references_1 = require('./payment-references.js');
Object.defineProperty(exports, 'PaymentReferencesPage', {
  enumerable: true,
  get: function () {
    return payment_references_1.PaymentReferencesPage;
  },
});
Object.defineProperty(exports, 'PaymentReferences', {
  enumerable: true,
  get: function () {
    return payment_references_1.PaymentReferences;
  },
});
var returns_1 = require('./returns.js');
Object.defineProperty(exports, 'ReturnObjectsPage', {
  enumerable: true,
  get: function () {
    return returns_1.ReturnObjectsPage;
  },
});
Object.defineProperty(exports, 'Returns', {
  enumerable: true,
  get: function () {
    return returns_1.Returns;
  },
});
var routing_details_1 = require('./routing-details.js');
Object.defineProperty(exports, 'RoutingDetailsPage', {
  enumerable: true,
  get: function () {
    return routing_details_1.RoutingDetailsPage;
  },
});
Object.defineProperty(exports, 'RoutingDetails', {
  enumerable: true,
  get: function () {
    return routing_details_1.RoutingDetails;
  },
});
var validations_1 = require('./validations.js');
Object.defineProperty(exports, 'Validations', {
  enumerable: true,
  get: function () {
    return validations_1.Validations;
  },
});
var transactions_1 = require('./transactions/transactions.js');
Object.defineProperty(exports, 'TransactionsPage', {
  enumerable: true,
  get: function () {
    return transactions_1.TransactionsPage;
  },
});
Object.defineProperty(exports, 'Transactions', {
  enumerable: true,
  get: function () {
    return transactions_1.Transactions;
  },
});
var virtual_accounts_1 = require('./virtual-accounts.js');
Object.defineProperty(exports, 'VirtualAccountsPage', {
  enumerable: true,
  get: function () {
    return virtual_accounts_1.VirtualAccountsPage;
  },
});
Object.defineProperty(exports, 'VirtualAccounts', {
  enumerable: true,
  get: function () {
    return virtual_accounts_1.VirtualAccounts;
  },
});
var webhooks_1 = require('./webhooks.js');
Object.defineProperty(exports, 'Webhooks', {
  enumerable: true,
  get: function () {
    return webhooks_1.Webhooks;
  },
});
