// modern-treasury/resources/ledger-accounts.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.LedgerAccountsPage = exports.LedgerAccounts = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class LedgerAccounts extends resource_1.APIResource {
  /**
   * Create a ledger account.
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/ledger_accounts', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  retrieve(id, query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.retrieve(id, {}, query);
    }
    return this.get(`/api/ledger_accounts/${id}`, { query, ...options });
  }
  update(id, body = {}, options) {
    if ((0, core_1.isRequestOptions)(body)) {
      return this.update(id, {}, body);
    }
    return this.patch(`/api/ledger_accounts/${id}`, { body, ...options });
  }
  list(query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/ledger_accounts', LedgerAccountsPage, { query, ...options });
  }
  /**
   * Delete a ledger account.
   */
  del(id, options) {
    return this.delete(`/api/ledger_accounts/${id}`, options);
  }
}
exports.LedgerAccounts = LedgerAccounts;
class LedgerAccountsPage extends pagination_1.Page {}
exports.LedgerAccountsPage = LedgerAccountsPage;
(function (LedgerAccounts) {})((LedgerAccounts = exports.LedgerAccounts || (exports.LedgerAccounts = {})));
