// modern-treasury/resources/ledger-event-handlers.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.LedgerEventHandlerListResponsesPage = exports.LedgerEventHandlers = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class LedgerEventHandlers extends resource_1.APIResource {
  /**
   * create ledger_event_handler
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/ledger_event_handlers', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * Get details on a single ledger event handler.
   */
  retrieve(id, options) {
    return this.get(`/api/ledger_event_handlers/${id}`, options);
  }
  list(query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/ledger_event_handlers', LedgerEventHandlerListResponsesPage, {
      query,
      ...options,
    });
  }
  /**
   * Archive a ledger event handler.
   */
  del(id, options) {
    return this.delete(`/api/ledger_event_handlers/${id}`, options);
  }
}
exports.LedgerEventHandlers = LedgerEventHandlers;
class LedgerEventHandlerListResponsesPage extends pagination_1.Page {}
exports.LedgerEventHandlerListResponsesPage = LedgerEventHandlerListResponsesPage;
(function (LedgerEventHandlers) {})(
  (LedgerEventHandlers = exports.LedgerEventHandlers || (exports.LedgerEventHandlers = {})),
);
