// modern-treasury/resources/ledger-account-payouts.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.LedgerAccountPayoutsPage = exports.LedgerAccountPayouts = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class LedgerAccountPayouts extends resource_1.APIResource {
  /**
   * Create a ledger account payout.
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/ledger_account_payouts', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  update(id, body = {}, options) {
    if ((0, core_1.isRequestOptions)(body)) {
      return this.update(id, {}, body);
    }
    return this.patch(`/api/ledger_account_payouts/${id}`, { body, ...options });
  }
  list(query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/ledger_account_payouts', LedgerAccountPayoutsPage, { query, ...options });
  }
  /**
   * Get details on a single ledger account payout.
   */
  retireve(id, options) {
    return this.get(`/api/ledger_account_payouts/${id}`, options);
  }
}
exports.LedgerAccountPayouts = LedgerAccountPayouts;
class LedgerAccountPayoutsPage extends pagination_1.Page {}
exports.LedgerAccountPayoutsPage = LedgerAccountPayoutsPage;
(function (LedgerAccountPayouts) {})(
  (LedgerAccountPayouts = exports.LedgerAccountPayouts || (exports.LedgerAccountPayouts = {})),
);
