// modern-treasury/resources/ledger-event-handlers.mjs
// File generated from our OpenAPI spec by Stainless.
import { APIResource } from 'modern-treasury/resource';
import { isRequestOptions } from 'modern-treasury/core';
import { Page } from 'modern-treasury/pagination';
export class LedgerEventHandlers extends APIResource {
  /**
   * create ledger_event_handler
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/ledger_event_handlers', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * Get details on a single ledger event handler.
   */
  retrieve(id, options) {
    return this.get(`/api/ledger_event_handlers/${id}`, options);
  }
  list(query = {}, options) {
    if (isRequestOptions(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/ledger_event_handlers', LedgerEventHandlerListResponsesPage, {
      query,
      ...options,
    });
  }
  /**
   * Archive a ledger event handler.
   */
  del(id, options) {
    return this.delete(`/api/ledger_event_handlers/${id}`, options);
  }
}
export class LedgerEventHandlerListResponsesPage extends Page {}
(function (LedgerEventHandlers) {})(LedgerEventHandlers || (LedgerEventHandlers = {}));
