// modern-treasury/resources/ledger-account-payouts.mjs
// File generated from our OpenAPI spec by Stainless.
import { APIResource } from 'modern-treasury/resource';
import { isRequestOptions } from 'modern-treasury/core';
import { Page } from 'modern-treasury/pagination';
export class LedgerAccountPayouts extends APIResource {
  /**
   * Create a ledger account payout.
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/ledger_account_payouts', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  update(id, body = {}, options) {
    if (isRequestOptions(body)) {
      return this.update(id, {}, body);
    }
    return this.patch(`/api/ledger_account_payouts/${id}`, { body, ...options });
  }
  list(query = {}, options) {
    if (isRequestOptions(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/ledger_account_payouts', LedgerAccountPayoutsPage, { query, ...options });
  }
  /**
   * Get details on a single ledger account payout.
   */
  retireve(id, options) {
    return this.get(`/api/ledger_account_payouts/${id}`, options);
  }
}
export class LedgerAccountPayoutsPage extends Page {}
(function (LedgerAccountPayouts) {})(LedgerAccountPayouts || (LedgerAccountPayouts = {}));
