// modern-treasury/resources/ledgers.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.LedgersPage = exports.Ledgers = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class Ledgers extends resource_1.APIResource {
  /**
   * Create a ledger.
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/ledgers', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * Get details on a single ledger.
   */
  retrieve(id, options) {
    return this.get(`/api/ledgers/${id}`, options);
  }
  update(id, body = {}, options) {
    if ((0, core_1.isRequestOptions)(body)) {
      return this.update(id, {}, body);
    }
    return this.patch(`/api/ledgers/${id}`, { body, ...options });
  }
  list(query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/ledgers', LedgersPage, { query, ...options });
  }
  /**
   * Delete a ledger.
   */
  del(id, options) {
    return this.delete(`/api/ledgers/${id}`, options);
  }
}
exports.Ledgers = Ledgers;
class LedgersPage extends pagination_1.Page {}
exports.LedgersPage = LedgersPage;
(function (Ledgers) {})((Ledgers = exports.Ledgers || (exports.Ledgers = {})));
