// modern-treasury/resources/payment-flows.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.PaymentFlowsPage = exports.PaymentFlows = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class PaymentFlows extends resource_1.APIResource {
  /**
   * create payment_flow
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/payment_flows', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  retrieve(id, query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.retrieve(id, {}, query);
    }
    const { 'Idempotency-Key': idempotencyKey } = query;
    return this.get(`/api/payment_flows/${id}`, {
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * update payment_flow
   */
  update(id, params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.patch(`/api/payment_flows/${id}`, {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  list(query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/payment_flows', PaymentFlowsPage, { query, ...options });
  }
}
exports.PaymentFlows = PaymentFlows;
class PaymentFlowsPage extends pagination_1.Page {}
exports.PaymentFlowsPage = PaymentFlowsPage;
(function (PaymentFlows) {})((PaymentFlows = exports.PaymentFlows || (exports.PaymentFlows = {})));
