// modern-treasury/resources/returns.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.ReturnObjectsPage = exports.Returns = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class Returns extends resource_1.APIResource {
  /**
   * Create a return.
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/returns', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * Get a single return.
   */
  retrieve(id, options) {
    return this.get(`/api/returns/${id}`, options);
  }
  list(query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/returns', ReturnObjectsPage, { query, ...options });
  }
}
exports.Returns = Returns;
class ReturnObjectsPage extends pagination_1.Page {}
exports.ReturnObjectsPage = ReturnObjectsPage;
(function (Returns) {})((Returns = exports.Returns || (exports.Returns = {})));
