// modern-treasury/resources/returns.mjs
// File generated from our OpenAPI spec by Stainless.
import { APIResource } from 'modern-treasury/resource';
import { isRequestOptions } from 'modern-treasury/core';
import { Page } from 'modern-treasury/pagination';
export class Returns extends APIResource {
  /**
   * Create a return.
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/returns', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * Get a single return.
   */
  retrieve(id, options) {
    return this.get(`/api/returns/${id}`, options);
  }
  list(query = {}, options) {
    if (isRequestOptions(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/returns', ReturnObjectsPage, { query, ...options });
  }
}
export class ReturnObjectsPage extends Page {}
(function (Returns) {})(Returns || (Returns = {}));
