// modern-treasury/resources/virtual-accounts.mjs
// File generated from our OpenAPI spec by Stainless.
import { APIResource } from 'modern-treasury/resource';
import { isRequestOptions } from 'modern-treasury/core';
import { Page } from 'modern-treasury/pagination';
export class VirtualAccounts extends APIResource {
  /**
   * create virtual_account
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/virtual_accounts', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * get virtual_account
   */
  retrieve(id, options) {
    return this.get(`/api/virtual_accounts/${id}`, options);
  }
  update(id, body = {}, options) {
    if (isRequestOptions(body)) {
      return this.update(id, {}, body);
    }
    return this.patch(`/api/virtual_accounts/${id}`, { body, ...options });
  }
  list(query = {}, options) {
    if (isRequestOptions(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/virtual_accounts', VirtualAccountsPage, { query, ...options });
  }
  /**
   * delete virtual_account
   */
  del(id, options) {
    return this.delete(`/api/virtual_accounts/${id}`, options);
  }
}
export class VirtualAccountsPage extends Page {}
(function (VirtualAccounts) {})(VirtualAccounts || (VirtualAccounts = {}));
