// modern-treasury/pagination.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.Page = void 0;
const core_1 = require('./core.js');
class Page extends core_1.AbstractPage {
  constructor(client, response, options) {
    super(client, response, options);
    this.items = response || [];
    this.per_page = (0, core_1.coerceInteger)(response.responseHeaders['x-per-page'] || null);
    this.after_cursor = response.responseHeaders['x-after-cursor'] || null;
  }
  getPaginatedItems() {
    return this.items;
  }
  // @deprecated Please use `nextPageInfo()` instead
  nextPageParams() {
    const info = this.nextPageInfo();
    if (!info) return null;
    if ('params' in info) return info.params;
    const params = Object.fromEntries(info.url.searchParams);
    if (!Object.keys(params).length) return null;
    return params;
  }
  nextPageInfo() {
    if (!this.after_cursor) return null;
    return { params: { after_cursor: this.after_cursor } };
  }
}
exports.Page = Page;
