// modern-treasury/resources/invoices/invoices.mjs
// File generated from our OpenAPI spec by Stainless.
import { APIResource } from 'modern-treasury/resource';
import { isRequestOptions } from 'modern-treasury/core';
import { LineItems } from './line-items.mjs';
import * as API from './index.mjs';
import { Page } from 'modern-treasury/pagination';
export class Invoices extends APIResource {
  constructor() {
    super(...arguments);
    this.lineItems = new LineItems(this.client);
  }
  /**
   * create invoice
   */
  create(params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post('/api/invoices', {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * get invoice
   */
  retrieve(id, options) {
    return this.get(`/api/invoices/${id}`, options);
  }
  update(id, body = {}, options) {
    if (isRequestOptions(body)) {
      return this.update(id, {}, body);
    }
    return this.patch(`/api/invoices/${id}`, { body, ...options });
  }
  list(query = {}, options) {
    if (isRequestOptions(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/invoices', InvoicesPage, { query, ...options });
  }
}
export class InvoicesPage extends Page {}
(function (Invoices) {
  Invoices.LineItems = API.LineItems;
  Invoices.InvoiceLineItemsPage = API.InvoiceLineItemsPage;
})(Invoices || (Invoices = {}));
