// modern-treasury/resources/invoices/line-items.js
'use strict';
// File generated from our OpenAPI spec by Stainless.
Object.defineProperty(exports, '__esModule', { value: true });
exports.InvoiceLineItemsPage = exports.LineItems = void 0;
const resource_1 = require('modern-treasury/resource');
const core_1 = require('modern-treasury/core');
const pagination_1 = require('modern-treasury/pagination');
class LineItems extends resource_1.APIResource {
  /**
   * create invoice_line_item
   */
  create(invoiceId, params, options) {
    const { 'Idempotency-Key': idempotencyKey, ...body } = params;
    return this.post(`/api/invoices/${invoiceId}/invoice_line_items`, {
      body,
      ...options,
      headers: {
        'Idempotency-Key': idempotencyKey || '',
        ...(options === null || options === void 0 ? void 0 : options.headers),
      },
    });
  }
  /**
   * get invoice_line_item
   */
  retrieve(invoiceId, id, options) {
    return this.get(`/api/invoices/${invoiceId}/invoice_line_items/${id}`, options);
  }
  update(invoiceId, id, body = {}, options) {
    if ((0, core_1.isRequestOptions)(body)) {
      return this.update(invoiceId, id, {}, body);
    }
    return this.patch(`/api/invoices/${invoiceId}/invoice_line_items/${id}`, { body, ...options });
  }
  list(invoiceId, query = {}, options) {
    if ((0, core_1.isRequestOptions)(query)) {
      return this.list(invoiceId, {}, query);
    }
    return this.getAPIList(`/api/invoices/${invoiceId}/invoice_line_items`, InvoiceLineItemsPage, {
      query,
      ...options,
    });
  }
  /**
   * delete invoice_line_item
   */
  del(invoiceId, id, options) {
    return this.delete(`/api/invoices/${invoiceId}/invoice_line_items/${id}`, options);
  }
}
exports.LineItems = LineItems;
class InvoiceLineItemsPage extends pagination_1.Page {}
exports.InvoiceLineItemsPage = InvoiceLineItemsPage;
(function (LineItems) {})((LineItems = exports.LineItems || (exports.LineItems = {})));
