// modern-treasury/resources/transactions/transactions.mjs
// File generated from our OpenAPI spec by Stainless.
import { APIResource } from 'modern-treasury/resource';
import { isRequestOptions } from 'modern-treasury/core';
import { LineItems } from './line-items.mjs';
import * as API from './index.mjs';
import { Page } from 'modern-treasury/pagination';
export class Transactions extends APIResource {
  constructor() {
    super(...arguments);
    this.lineItems = new LineItems(this.client);
  }
  /**
   * Get details on a single transaction.
   */
  retrieve(id, options) {
    return this.get(`/api/transactions/${id}`, options);
  }
  update(id, body = {}, options) {
    if (isRequestOptions(body)) {
      return this.update(id, {}, body);
    }
    return this.patch(`/api/transactions/${id}`, { body, ...options });
  }
  list(query = {}, options) {
    if (isRequestOptions(query)) {
      return this.list({}, query);
    }
    return this.getAPIList('/api/transactions', TransactionsPage, { query, ...options });
  }
}
export class TransactionsPage extends Page {}
(function (Transactions) {
  Transactions.LineItems = API.LineItems;
  Transactions.TransactionLineItemsPage = API.TransactionLineItemsPage;
})(Transactions || (Transactions = {}));
